src/app/settings/keyboard-mapping-settings/components/keyboard-mapping-settings-page/keyboard-mapping-settings-page.component.ts

```diff
@@ -0,0 +1,221 @@
+        }
+        this.actionTriggerService.updateActionTrigger(result.actionTrigger).subscribe()
+        this.notificationService.createInfoNotification('Successfully updated Keyboard Mapping')
+      },
+      keyboardMapping
+    )
+  }
+
+  public getActionTriggersFromKeyboardMappings(): ActionTrigger<KeyboardTriggerData>[] {
+    return this.keyboardMappings.map(keyboardMapping => keyboardMapping.actionTrigger)
+  }
+
+  public actionTriggersUploaded(actionTriggers: ActionTrigger<KeyboardTriggerData>[]): void {
+    Promise.all(
+      actionTriggers.map(actionTrigger => {
+        const actionTriggerAlreadyExist: boolean = this.keyboardMappings.some(actionTriggerWithAction => actionTriggerWithAction.actionTrigger.id === actionTrigger.id)
+        actionTriggerAlreadyExist ? this.actionTriggerService.updateActionTrigger(actionTrigger).subscribe() : this.actionTriggerService.createActionTrigger(actionTrigger).subscribe()
+      })
+    )
+      .then(() => this.notificationService.createInfoNotification('Successfully imported Keyboard Mappings'))
+      .catch(error => {
+        this.logger.error(error)
+        this.notificationService.createErrorNotification('Something went wrong uploading Keyboard Mappings!')
+      })
+  }
+
+  public validateActionTriggers(actionTriggers: ActionTrigger<KeyboardTriggerData>[]): boolean {
+    try {
+      return !!this.actionTriggerParser.parseActionTriggers(actionTriggers)
+    } catch {
+      return false
+    }
+  }
+
+  public deleteSelectedKeyboardMapping(): void {
+    this.dialogService.createConfirmDialog($localize`global.delete.label`, $localize`keyboard-mapping.delete-selected.confirmation`, $localize`global.delete.label`, () => {
+      const actionTriggerIdsToBeDeleted: string[] = this.keyboardMappings
+        .filter(keyboardMapping => this.selectedKeyboardMappings.has(keyboardMapping))
+        .map(keyboardMapping => keyboardMapping.actionTrigger.id)
+
+      this.selectedKeyboardMappings.clear()
+
+      Promise.all(actionTriggerIdsToBeDeleted.map(id => this.actionTriggerService.deleteActionTrigger(id).subscribe()))
+        .then(() => this.notificationService.createInfoNotification('Successfully deleted all selected Keyboard Mappings'))
+        .catch(error => {
+          this.logger.error(error)
+          this.notificationService.createErrorNotification('Something went wrong deleting all selected Keyboard Mappings!')
+        })
+    })
+  }
+
+  public doesKeyboardMappingMatchSearchQuery(keyboardMapping: KeyboardMapping): boolean {
+    if (!this.keyboardMappingSearchQuery || this.keyboardMappingSearchQuery.length === 0) {
+      return true
+    }
+
+    const lowerCasedQuery: string = this.keyboardMappingSearchQuery.toLowerCase()
+    const doesLabelMatchQuery: boolean = keyboardMapping.actionTrigger.data.label.toLowerCase().includes(lowerCasedQuery)
+    const doesShortCutMatchQuery: boolean = this.getShortcutName(keyboardMapping).toLowerCase().includes(lowerCasedQuery)
```

Comment:
Since shortcut is one word this should be `doesShortcutMatchQuery`

Reply:
Updated